### PR TITLE
Deprecation warning to repos without renovate and default settings

### PR DESCRIPTION
**Pull request by adusumillipraveen**
Notes:
* Tested in https://github.com/hmcts/cnp-plum-recipes-service/pull/822/files 
* 
Screenshot 2023-08-23 at 10 16 46

### CODE REVIEW

**Review by thomast1906 — APPROVED** [review]
Nice! Want to announce this?

File: resources/uk/gov/hmcts/renovate/renovate-config-check.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -x
+
+if [[ ! -f .github/renovate.json ]]
```
Comment (timja) [inline resources/uk/gov/hmcts/renovate/renovate-config-check.sh:4]: i wouldn't enforce this location, renovate can be in a number of places, I would at least support .github/renovate.json and renovate.json.

If renovate automatically onboarded a team by sending them a PR it does it to renovate.json

File: vars/sectionBuildAndTest.groovy [hunk 2]
```diff
@@ -36,6 +36,8 @@ def call(params) {
     }
     builder.setupToolVersion()
   }
+  echo("test")
```
Comment (timja) [inline vars/sectionBuildAndTest.groovy:39]:
```suggestion
```

File: vars/withInfraPipeline.groovy [hunk 3]
```diff
@@ -37,7 +37,7 @@ def call(String product, String component = null, Closure body) {
       stageWithAgent('Checkout', product) {
         checkoutScm(pipelineCallbacksRunner: callbacksRunner)
       }
-
+      warnAboutRenovateConfig()
```
Comment (timja) [inline vars/withInfraPipeline.groovy:40]: couldn't we put this in checkoutScm? it seems like you've had to put this in a lot of places

File: resources/uk/gov/hmcts/renovate/renovate-config-check.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -x
+
+FILE_PATH=""
+if [[ -f .github1/renovate.json ]]; then
```
Comment (timja) [inline resources/uk/gov/hmcts/renovate/renovate-config-check.sh:5]:
```suggestion
if [[ -f .github/renovate.json ]]; then
```

**Review by timja — APPROVED** [review]

**Pull request merged**